### PR TITLE
Added schema for `/_plugins/_knn/stats`, `/_plugins/_knn/models/{model_id}`, `_train` and `_search`.

### PR DESCRIPTION
**Pull request by dblock**
### Description

Added schema for `/_plugins/_knn/stats`, `/_plugins/_knn/models/{model_id}`, `_train` and `_search`.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: spec/schemas/knn._common.yaml [hunk 1]
```diff
@@ -25,3 +25,164 @@ components:
...
+          $ref: '_common.yaml#/components/schemas/DurationValueUnitMillis'
+    GraphMergeStats:
+      type: object
+      properties:
+        current:
+          type: number
+        total:
+          type: number
+        total_time_in_millis:
+          type: number
```
Comment (Tokesh) [inline spec/schemas/knn._common.yaml:137]: I think we can also use ref:
```  $ref: '_common.yaml#/components/schemas/DurationValueUnitMillis' ```

File: spec/schemas/knn._common.yaml [hunk 2]
```diff
@@ -25,3 +25,164 @@ components:
...
+        current_docs:
+          type: number
+        total_docs:
+          type: number
+        total_size_in_bytes:
+          type: number
```
Comment (Tokesh) [inline spec/schemas/knn._common.yaml:143]: Also here: ```$ref: '../schemas/_common.yaml#/components/schemas/ByteCount' ```

File: spec/schemas/knn._common.yaml [hunk 3]
```diff
@@ -25,3 +25,164 @@ components:
...
+        current_size_in_bytes:
+          type: number
```
Comment (Tokesh) [inline spec/schemas/knn._common.yaml:145]: ```$ref: '../schemas/_common.yaml#/components/schemas/ByteCount'```

**Review by Tokesh — COMMENTED** [review]
Hi everyone! I wanted to review a PR in Opensearch for the first time. Everything looks great, but I noticed a couple of things that I have encountered in my PRs.

**Review by Xtansia — APPROVED** [review]

**Pull request merged**